src/main/java/com/google/cloud/tools/maven/AppEngineStandardDeployer.java

```diff
@@ -19,177 +19,190 @@
   }
 
   @VisibleForTesting
-  AppEngineStandardDeployer(AbstractDeployMojo deployMojo, AppEngineStager stager) {
+  AppEngineStandardDeployer(AbstractDeployMojo deployMojo, AppEngineStandardStager stager) {
     this.deployMojo = deployMojo;
     this.stager = stager;
+
+    stager.overrideAppEngineDirectory();
+    setDeploymentProjectAndVersion();
   }
 
   @Override
-  public void deploy() throws MojoFailureException, MojoExecutionException {
+  public void deploy() throws MojoExecutionException {
     stager.stage();
-    deployMojo.deployables.clear();
-    deployMojo.deployables.add(deployMojo.stagingDirectory);
+    deployMojo.setDeployables(ImmutableList.of(deployMojo.getStagingDirectory()));
 
     try {
-      updatePropertiesFromAppEngineWebXml();
       deployMojo.getAppEngineFactory().deployment().deploy(deployMojo);
-    } catch (AppEngineException | SAXException | IOException ex) {
-      throw new MojoFailureException(ex.getMessage(), ex);
+    } catch (AppEngineException ex) {
+      throw new MojoExecutionException("Standard application deployment failed", ex);
     }
   }
 
   @Override
-  public void deployAll() throws MojoExecutionException, MojoFailureException {
+  public void deployAll() throws MojoExecutionException {
     stager.stage();
-    deployMojo.deployables.clear();
+    ImmutableList.Builder<File> standardDeployables = ImmutableList.builder();
 
     // Look for app.yaml
-    File appYaml = deployMojo.stagingDirectory.toPath().resolve("app.yaml").toFile();
+    File appYaml = deployMojo.getStagingDirectory().toPath().resolve("app.yaml").toFile();
     if (!appYaml.exists()) {
       throw new MojoExecutionException("Failed to deploy all: could not find app.yaml.");
     }
     deployMojo.getLog().info("deployAll: Preparing to deploy app.yaml");
-    deployMojo.deployables.add(appYaml);
+    standardDeployables.add(appYaml);
 
     // Look for config yamls
     String[] configYamls = {"cron.yaml", "dispatch.yaml", "dos.yaml", "index.yaml", "queue.yaml"};
     Path configPath =
-        deployMojo.stagingDirectory.toPath().resolve("WEB-INF").resolve("appengine-generated");
+        deployMojo.getStagingDirectory().toPath().resolve("WEB-INF").resolve("appengine-generated");
     for (String yamlName : configYamls) {
       File yaml = configPath.resolve(yamlName).toFile();
       if (yaml.exists()) {
         deployMojo.getLog().info("deployAll: Preparing to deploy " + yamlName);
-        deployMojo.deployables.add(yaml);
+        standardDeployables.add(yaml);
       }
     }
 
+    deployMojo.setDeployables(standardDeployables.build());
+
     try {
-      updatePropertiesFromAppEngineWebXml();
       deployMojo.getAppEngineFactory().deployment().deploy(deployMojo);
-    } catch (AppEngineException | SAXException | IOException ex) {
-      throw new RuntimeException(ex);
+    } catch (AppEngineException ex) {
+      throw new MojoExecutionException("Failed to Deploy", ex);
```

Comment:
Missed a few of these (`deployAll`, `deployCron`, `deployDispatch`, `deployDos` in `AppEngineStandardDeployer`)

Reply:
ah.. I'll go fix those.